crates/bevy_ecs/src/reflect/entity_commands.rs

```diff
@@ -198,12 +202,16 @@ fn insert_reflect(
         panic!("error[B0003]: Could not insert a reflected component (of type {type_path}) for entity {entity:?} because it doesn't exist in this World. See: https://bevyengine.org/learn/errors/b0003");
     };
     let Some(type_registration) = type_registry.get(type_info.type_id()) else {
-        panic!("Could not get type registration (for component type {type_path}) because it doesn't exist in the TypeRegistry.");
+        panic!("could not get type registration (for component type {type_path}) because it doesn't exist in the TypeRegistry.");
     };
-    let Some(reflect_component) = type_registration.data::<ReflectComponent>() else {
-        panic!("Could not get ReflectComponent data (for component type {type_path}) because it doesn't exist in this TypeRegistration.");
-    };
-    reflect_component.insert(&mut entity, component.as_partial_reflect(), type_registry);
+
+    if let Some(reflect_component) = type_registration.data::<ReflectComponent>() {
+        reflect_component.insert(&mut entity, component.as_partial_reflect(), type_registry);
+    } else if let Some(reflect_bundle) = type_registration.data::<ReflectBundle>() {
+        reflect_bundle.insert(&mut entity, component.as_partial_reflect(), type_registry);
+    } else {
+        panic!("`{type_path}` should have #[reflect(Component)] or #[reflect(Bundle)].");
```

Comment:
Panic messages don't need to have dots. The same in the message above.
```suggestion
        panic!("`{type_path}` should have #[reflect(Component)] or #[reflect(Bundle)]");
```